services/asanaService.js

```diff
@@ -0,0 +1,41 @@
+import dbConnect from "../lib/dbConnect";
+import Asana from "../models/Asana";
+
+export async function getAllAsanas() {
+  await dbConnect();
+
+  const asanas = await Asana.find();
+  console.log(asanas);
+  const sanatizedAsanas = asanas.map((asana) => ({
+    id: asana._id,
+    english_name: asana.english_name,
+    sanskrit_name: asana.sanskrit_name,
+    img_url: asana.img_url,
+    description: asana.description,
+    levels: asana.levels,
+    category: asana.category,
+    benefit: asana.benefit,
+    next: asana.next,
+  }));
+
+  return JSON.parse(JSON.stringify(sanatizedAsanas));
+}
+
+export async function getAsanaById(id) {
+  await dbConnect();
+  const asanas = await Asana.find();
+  const asana = asanas.find((asana) => asana.id === id);
```

Comment:
There's an inbuilt function in mongoose called `findById` which you can use to directly pass down the id. But be aware, you are returned an object as a result, not an array.

Reply:
Done